frontend/server/src/Controllers/Certificate.php

```diff
@@ -415,6 +415,111 @@ private static function getCoderOfTheMonthCertificate(
         );
     }
 
+    /**
+     * Generates all the certificates for a contest given its contest ID.
+     *
+     * @throws \OmegaUp\Exceptions\NotFoundException
+     *
+     * @return array{status: string}
+     *
+     * @omegaup-request-param int|null $certificates_cutoff
+     * @omegaup-request-param int|null $contest_id
+     */
+    public static function apiGenerateContestCertificates(\OmegaUp\Request $r) {
+        \OmegaUp\Controllers\Controller::ensureNotInLockdown();
+
+        $r->ensureMainUserIdentity();
+
+        $contestID = $r->ensureInt('contest_id');
+
+        // obtain the contest
+        $contest = \OmegaUp\DAO\Contests::getByPK($contestID);
+
+        if (
+            is_null($contest)
+            || is_null($contest->problemset_id)
+            || is_null($contest->alias)
+        ) {
+            throw new \OmegaUp\Exceptions\NotFoundException('contestNotFound');
+        }
+
+        $problemset = \OmegaUp\DAO\Problemsets::getByPK(
+            $contest->problemset_id
+        );
+
+        if (is_null($problemset)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('contestNotFound');
+        }
+
+        // check whether the logged user is a certificate generator
+        if (!\OmegaUp\Authorization::isCertificateGenerator($r->identity)) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException();
+        }
+
+        if ($contest->certificates_status !== 'uninitiated' && $contest->certificates_status !== 'retryable_error') {
+            return ['status' => 'ok'];
+        }
+
+        $certificateCutoff = $r->ensureOptionalInt('certificates_cutoff');
+
+        // add certificates_cutoff value to the course
+        if (!is_null($certificateCutoff)) {
+            $contest->certificate_cutoff = $certificateCutoff;
+        }
+
+        // update contest with the new value
+        \OmegaUp\DAO\Contests::update($contest);
+
+        // get contest info
+        $contestExtraInformation = \OmegaUp\DAO\Contests::getByAliasWithExtraInformation(
+            $contest->alias
+        );
+
+        if (is_null($contestExtraInformation)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('contestNotFound');
+        }
+
+        // set RabbitMQ client parameters
+        $routingKey = 'ContestQueue';
+        $exchange = 'certificates';
+
+        //connection to rabbitmq
+        $channel = \OmegaUp\RabbitMQConnection::getInstance()->channel();
+
+        $channel->exchange_declare($exchange, 'direct', false, true, true);
```

Comment:
Acá sería bueno indicar el nombre de cada uno de los parámetros para saber a qué corresponde cada uno:
```suggestion
        $channel->exchange_declare(
            $exchange,
            type: 'direct',
            passive: false,
            durable: true,
            auto_delete: false
        );
```

Si te fijas, cambié el valor de `auto_delete` a `false` debido a que así lo tenemos configurado en python y con eso evitamos posibles fallas en las pre-condiciones